src/test_lib_json/jsontest.cpp

```diff
@@ -275,13 +273,12 @@ bool Runner::runAllTest(bool printSummary) const {
     }
     return true;
   } else {
-    for (unsigned int index = 0; index < failures.size(); ++index) {
-      TestResult& result = failures[index];
+    for (const TestResult& result : failures) {
```

Comment:
You're adding constness here, and printFailure sounds like it might be or soon become non-const. let's leave off the const.